### ISSUE TITLE
DBFCDX problem with sorting and special characters

### ISSUE THREAD

**Issue by cpyrgas** [X-Sharp/XSharpPublic#104]
In the sample below, a value is put into a record, but then seeking for this exact value returns FALSE. The problem seems to happen with specific contents of records and only when there's a call to DBCommit(). Removing this, it works fine.

```
FUNCTION Start( ) AS VOID
	LOCAL cDBF AS STRING 
	LOCAL aFields, aValues AS ARRAY 
	LOCAL i AS DWORD 
	
	RddSetDefault ( "DBFCDX" ) 
	
	// makes no difference
//	SetCollation ( #clipper ) 
//	SetCollation ( #windows ) 
	
	cDBF := "c:\test\aaa"
	FErase ( cDBF + IndexExt() )	
	
	aFields := { { "LAST" , "C" , 20 , 0 } } 
	aValues := { "Art" ,"Aero" , "Anfang",  "Goethe" , "Äffin" , ;
				"Ärger" , "Ärmlich", "Goldmann" ,"Üblich" , "Göthe" }  
	
	DbCreate( cDBF , AFields)
	DbUseArea( ,,cDBF , , FALSE )		
	
	FOR i := 1 UPTO ALen ( aValues )
		DbAppend() 
		FieldPut ( 1 , aValues [ i ] ) 
	NEXT         
	
	DbCreateOrder ( "ORDER1" , cDBF , "upper(LAST)" , { || Upper ( _Field->LAST) } )
	DbGoTop()
	
	// ? "Seek GOETHE" , DbSeek ( "GOETHE")
	DbGoto(4) // same result as with seek()
	FieldPut ( 1 , "GOOETHE" ) 
	? RecNo() // 4
	? DbCommit() // without this, it works correctly!
	?
	DbGoTop()
	? "Seek GOOETHE" , DbSeek ( "GOOETHE") // returns false, it should find it
	
	DbGoto(4)
	? "Record 4 has", FieldGet(1)
	?
	
	wait
	DbGoTop()
	?
	? "OrdKeyCount()", OrdKeyCount()
	? "Reccount()" , RecCount()
	?
	
	DO WHILE ! Eof() 
		? FieldGet ( 1 ) 
		DbSkip ( 1 ) 
	ENDDO 	
	DbCloseAll() 	
RETURN
```

**Comment by cpyrgas** [comment]
[VSSample1.zip](https://github.com/X-Sharp/XSharpPublic/files/3424846/VSSample1.zip)

**Comment by cpyrgas** [comment]
This is also the original sample by Karl-Heinz. All seek operations must return TRUE.

```
FUNCTION Start( ) AS VOID
	LOCAL cDBF, cPfad, cDriver, cIndex AS STRING 
	LOCAL aFields, aValues AS ARRAY 
	LOCAL i AS DWORD 
	LOCAL lFound, lAddUmlauteName,lUseGermanDll AS LOGIC 
	
	SetAnsi ( TRUE ) 
	SetCentury( TRUE )   
	
	cDriver := RddSetDefault ( "DBFCDX" ) 
	
    // ---------------- 
	lUseGermanDll := FALSE
	// ------------- 
	
	IF lUseGermanDll
		
		SetInternational ( #clipper  ) 
		SetCollation ( #clipper ) 
		
		IF ! SetNatDLL ( "German" )  		   
			? "German not loaded" 
		ENDIF 	                           
		SetDateCountry ( DateCountry.German )
		SetThousandSep ( Asc ( "." ))   
		SetDecimalSep ( Asc ( "," ))   
	ELSE 
//    	SetAppLocaleID(MAKELCID(MAKELANGID(LANG_GERMAN, SUBLANG_GERMAN), SORT_GERMAN_PHONE_BOOK))  // Telefonbuch    	
//		SetAppLocaleID(MAKELCID(MAKELANGID(LANG_GERMAN, SUBLANG_GERMAN), SORT_DEFAULT)) // Wörterbuch
	ENDIF  
	
	cDBF := cPfad + "Foo"
	cIndex := cPfad + "Foox" 
	
	FErase ( cIndex + IndexExt() )	
	
	aFields := { { "LAST" , "C" , 20 , 0 } } 
	
	aValues := { "Art" ,"Aero" , "Anfang",  "Goethe" , "Äffin" , "Ärger" , "Ärmlich", "Goldmann" ,;
				"Götz" , "Ober" , "Otter" , "Österreich" , "Östrogon" , "Ötzi" ,"Unter" , ;
				"Übel" , "Überheblich" , "Üblich" , "Göthe" }  
	
	// -------------
	
	lAddUmlauteName := TRUE 
	
	IF lAddUmlauteName				
		AAdd( aValues , "Göbel")
	ELSE 
		AAdd( aValues , "Gobel")
	ENDIF 		
	
    // -------------------
	
	? DbCreate( cDBF , AFields)
	? DbUseArea( ,,cDBF , , TRUE )		
	
	FOR i := 1 UPTO ALen ( aValues )
		DbAppend() 
		FieldPut ( 1 , aValues [ i ] ) 
	NEXT         
	
	? DbCreateOrder ( "ORDER1" , cIndex , "upper(LAST)" , { || Upper ( _Field->LAST) } )
	
	? DbSetOrder ( 1 )  
	
	DbGoTop()
	
	?
	? "Seek ART", DbSeek ( "ART") 
	? DbRLock()
	FieldPut ( 1 , "Aart" ) 
	? DbCommit()
	? DbRUnLock()
	?
	? "Seek AART" , DbSeek ( "AART")
	? DbRLock()
	FieldPut ( 1 , "Art" ) 	
	? DbCommit()
	? DbRUnLock()
	?
	? "Seek GOETHE" , DbSeek ( "GOETHE")
	? DbRLock()
	FieldPut ( 1 , "Gooethe" ) 
	? DbCommit()
	? DbRUnLock()
	?
	? "Seek GOOETHE" , lFound:= DbSeek ( "GOOETHE")
	IF lFound
		? FieldGet ( 1 )
		? DbRLock()
		FieldPut ( 1 , "Goethe" ) 	
		? DbCommit()
		? DbRUnLock()
	ELSE 
		? "Not found " , Eof()
	ENDIF 		
	
	DbGoTop()
	
	?
	? "OrdKeyCount()", OrdKeyCount()
	? "Reccount()" , RecCount()
	?
	
/*	DO WHILE ! Eof() 
		
		? FieldGet ( 1 ) 
		DbSkip ( 1 ) 
		
	ENDDO 	
*/	
	
	DbCloseAll() 	
	RddSetDefault ( cDriver )
RETURN
```

[VSSample_orig.zip](https://github.com/X-Sharp/XSharpPublic/files/3424856/VSSample_orig.zip)

**Comment by RobertvanderHulst** [comment]
Fixed with dbeeec76e3ef5ade2f01a06f954259bd54d00e69
